R/MSnExperiment-class.R

```diff
@@ -762,6 +1328,40 @@ setMethod("[[", "MSnExperiment",
               x[i, , drop = TRUE]
           })
 
+#' @rdname MSnExperiment
+setMethod("filterAcquisitionNum", "MSnExperiment", function(object, n, file) {
+    if (missing(n)) return(object)
+    if (missing(file))
+        file <- unique(fromFile(object))
+    if (!is.integer(n))
+        stop("'n' has to be an integer representing the acquisition number(s)",
+             " for sub-setting.")
+    if (!is.integer(file))
+        stop("'file' has to be an integer with the index of the file(s)",
+             " for sub-setting.")
+    sel_file <- fromFile(object) %in% file
+    sel_acq <- acquisitionNum(object) %in% n & sel_file
+    if (!any(sel_acq))
+        warning("No spectra with the specified acquisition number(s) found.")
+    object <- object[sel_acq | !sel_file, drop = FALSE]
+    object@processing <- c(object@processing,
+                           paste0("Filter: select by: ", length(n),
+                                  " acquisition number(s) in ", length(file),
+                                  " file(s). [", date(), "]"))
+    object
+})
+
+#' @rdname MSnExperiment
+setMethod("filterEmptySpectra", "MSnExperiment", function(object) {
+    empties <- isEmpty(object)
+    if (any(empties))
+        object <- object[!empties, drop = FALSE]
+    object@processing <- c(object@processing,
```

Comment:
I also thought about that - but then I opted against it because I felt it a little overdone. In the end it's a simple `paste` call and if we are careful we will be able to avoid differences in white spaces etc. For now I would leave it like it is - also because it's not clear if we keep the `@processing` slot in this class.